src/main/scala/com/zeta/playthegame/repository/GameAppointmentRepository.scala

```diff
@@ -1,40 +1,49 @@
 package com.zeta.playthegame.repository
 
+import cats.effect.IO
 import com.zeta.playthegame.GameAppointmentRequest
 import com.zeta.playthegame.repository.Entities._
 import com.zeta.playthegame.util.LoggerPerClassAware
 import org.mongodb.scala.Completed
 import org.mongodb.scala.bson.ObjectId
 import org.mongodb.scala.bson.collection.immutable.Document
 
-import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.ExecutionContext
 
-object GameAppointmentRepository extends LoggerPerClassAware {
-  import MongoConnection._
+class GameAppointmentRepository(mongoConnection: MongoConnection)(implicit executionContext: ExecutionContext) extends LoggerPerClassAware {
```

Comment:
Me explicas que es el `LoggerPerClassAware` ?

Reply:
hdp, `FindFile -> LoggerPerClassAware`

Reply:
Si, lo hice. Pero no entiendo para que lo necesitas. Quiero que me expliques bbsito

Reply:
Para loggear, sin tener que andar declarando a cada rato un Logger en cada clase

Reply:
Sigo sin entender, pero quizas mejor lo discutimos por slack. En nuestra implementacion funcional pura no veo que lo necesitemos. Cualquier error llega hasta el final del mundo y se loggea alli. Se me hace que estas metiendo efectos colaterales aca adentro, pero tampoco estoy muy seguro porque no veo ni una linea que diga `logger.info(...)`.

Reply:
Creo que entiendo el punto. Llegado el caso, si tenes que loggear algo, seguramente cats IO te probee de una "consola funcional".

Con ZIO harias: 

```scala
import scalaz.zio.console.putStrLn

val appointment =
  for {
  appointment <- findAppointment
  _ <- putStrLn(s"estoy loggeando el appointment $appointment")
} yield appointment
```
o si lo queres sin el for comprehension lo haces con flatMap o alguno de esos operadores falopa, creo que seria el `val appointment = findAppointment <* putStrLn`